### ISSUE TITLE
Consolidate user and org ids under the same namespace

### ISSUE THREAD

**Issue by NunoAlexandre** [radicle-dev/radicle-registry#526]
Currently, when registering a user, we check whether that user id is already taken under the user namespace:

``` rust
if store::Users::get(message.user_id.clone()).is_some() {
    return Err(RegistryError::DuplicateUserId.into())
}
```

The same goes for orgs.

### `TBD`

- The `spec` defines that user and org ids should live under the same namespace, meaning that, when registering either an org or a user, we need to check if no user nor org is registered with the given id

- Drop `DuplicateUserId` and `DuplicateOrgId` in favour of `IdAlreadyTaken`

-  Add unit tests covering this use case

**Comment by geigerzaehler** [comment]
We should also consider renaming the `DuplicateUserId` to better reflect the case.

**Comment by NunoAlexandre** [comment]
> We should also consider renaming the `DuplicateUserId` to better reflect the case.

Good point. I'd suggest `IdAlreadyTaken`.

**Comment by geigerzaehler** [comment]
Closed in #531
